src/http/client.cc

```diff
@@ -130,13 +130,30 @@ future<> connection::close() {
     return when_all(_read_buf.close(), _write_buf.close()).discard_result();
 }
 
+class basic_connection_factory : public connection_factory {
+    ipv4_addr _addr;
+public:
+    explicit basic_connection_factory(ipv4_addr addr)
+            : _addr(std::move(addr))
+    {
+    }
+    virtual future<connected_socket> make() override {
+        return seastar::connect(make_ipv4_address(_addr), {}, transport::TCP);
+    }
```

Comment:
Shouldn't the factory also expose an interface for restoring the connection? I guess you can wrap close(), but perhaps an symmetric interface is better.

Reply:
What is "restoring"? A client asks for a new connection and may in theory give the connection back to the factory, but then it's not a factory, but a pool itself

Reply:
I speculated about the implementation. I think the pool needs user control too. When are connections trimmed? Is the pool limited across all endpoints, or per-endpoint?

But maybe we don't need to start with everything.